lib/dry/configurable/test_interface.rb

```diff
@@ -8,11 +8,11 @@ module TestInterface
       #
       # @api public
       def reset_config
-        if self.is_a?(Class)
-          @config = _settings.create_config
-        else
-          @config = self.class._settings.create_config
-        end
+        @config = if [Class, Module].include? self.class
```

Comment:
`Module` is a superclass of `Class` so you only need to change it to `is_a?(Module)`, should be enough

Reply:
yes, this is better